File: frontend/src/store/api/result/dto/product/variant/HeadphonesVariantOutputDTO.ts
import { ProductVariantOutputDTO } from "./ProductVariantOutputDTO";

export interface HeadphonesVariantOutputDTO extends ProductVariantOutputDTO {
    nominalImpedance: number,
    loudness: number,
    color: string,
    headphonesId: number
};

export function isHeadphonesVariantOutputDTO(variant: ProductVariantOutputDTO): variant is HeadphonesVariantOutputDTO {
    return "headphonesId" in variant;
}

export function isHeadphonesVariantOutputDTOArray(variants: ProductVariantOutputDTO[]): variants is HeadphonesVariantOutputDTO[] {
    return variants.every(isHeadphonesVariantOutputDTO);
}

File: backend/product-client/src/main/java/org/andante/headphones/dto/HeadphonesVariantOutputDTO.java
package org.andante.headphones.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.andante.product.dto.ProductVariantOutputDTO;

@SuperBuilder
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class HeadphonesVariantOutputDTO extends ProductVariantOutputDTO {

    private Integer nominalImpedance;
    private Integer loudness;
    private String color;
    private Long headphonesId;
}
